### ISSUE TITLE
Refused to evaluate a string as JavaScript because 'unsafe-eval'

### ISSUE THREAD

**Issue by radarsu** [AtomLinter/linter-tslint#120]
linter-tslint: 0.11.1
typescript: 2.0.3

Information:
1. `tslint` works well on CLI.
2. Error pop ups on file save.
3. This error only appears in my angular2 project. In other typescript projects tslint works well.

```
EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".
    at Object.<anonymous> (C:\Users\Artur\Desktop\FTP - dunthag-client\node_modules\typescript\lib\typescript.js:60881:23)
    at Module._compile (C:\Users\Artur\AppData\Local\atom\app-1.11.2\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .js] (C:\Users\Artur\AppData\Local\atom\app-1.11.2\resources\app.asar\src\compile-cache.js:208:21)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (C:\Users\Artur\AppData\Local\atom\app-1.11.2\resources\app.asar\src\native-compile-cache.js:50:27)
    at Object.<anonymous> (C:\Users\Artur\Desktop\FTP - dunthag-client\node_modules\codelyzer\importDestructuringSpacingRule.js:7:10)
    at Module._compile (C:\Users\Artur\AppData\Local\atom\app-1.11.2\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .js] (C:\Users\Artur\AppData\Local\atom\app-1.11.2\resources\app.asar\src\compile-cache.js:208:21)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (C:\Users\Artur\AppData\Local\atom\app-1.11.2\resources\app.asar\src\native-compile-cache.js:50:27)
    at loadRule (C:\Users\Artur\.atom\packages\linter-tslint\node_modules\tslint\lib\ruleLoader.js:65:26)
    at findRule (C:\Users\Artur\.atom\packages\linter-tslint\node_modules\tslint\lib\ruleLoader.js:46:20)
    at Object.loadRules (C:\Users\Artur\.atom\packages\linter-tslint\node_modules\tslint\lib\ruleLoader.js:14:24)
    at Linter.lint (C:\Users\Artur\.atom\packages\linter-tslint\node_modules\tslint\lib\tslint.js:56:44)
    at file:///C:/Users/Artur/.atom/packages/linter-tslint/lib/init.coffee:103:31
```

**Comment by Arcanemagus** [comment]
The very odd thing about this is that we are already requiring `tslint` in a manner that should be excluding the possibility of this. Are you able to share code that triggers this?

The real solution here is for `tslint` to stop being terrible and using `eval` on code it is linting. 😛

**Comment by radarsu** [comment]
Here is a starter project I used: https://github.com/AngularClass/angular2-webpack-starter
And here is an actual project: https://github.com/radarsu/dunthag-client

The error occurrs in both projects, so it is not my-project-specific bug, but I am not currently able to find out which part of project could be causing this.

Stack from angular2-webpack-starter project:
```
EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".
    at Object.<anonymous> (C:\Users\Artur\Desktop\GitHub Repositories\angular2-webpack-starter\node_modules\typescript\lib\typescript.js:61632:23)
    at Module._compile (C:\Users\Artur\AppData\Local\atom\app-1.11.2\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .js] (C:\Users\Artur\AppData\Local\atom\app-1.11.2\resources\app.asar\src\compile-cache.js:208:21)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (C:\Users\Artur\AppData\Local\atom\app-1.11.2\resources\app.asar\src\native-compile-cache.js:50:27)
    at Object.<anonymous> (C:\Users\Artur\Desktop\GitHub Repositories\angular2-webpack-starter\node_modules\codelyzer\importDestructuringSpacingRule.js:7:10)
    at Module._compile (C:\Users\Artur\AppData\Local\atom\app-1.11.2\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .js] (C:\Users\Artur\AppData\Local\atom\app-1.11.2\resources\app.asar\src\compile-cache.js:208:21)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (C:\Users\Artur\AppData\Local\atom\app-1.11.2\resources\app.asar\src\native-compile-cache.js:50:27)
    at loadRule (C:\Users\Artur\.atom\packages\linter-tslint\node_modules\tslint\lib\ruleLoader.js:65:26)
    at findRule (C:\Users\Artur\.atom\packages\linter-tslint\node_modules\tslint\lib\ruleLoader.js:46:20)
    at Object.loadRules (C:\Users\Artur\.atom\packages\linter-tslint\node_modules\tslint\lib\ruleLoader.js:14:24)
    at Linter.lint (C:\Users\Artur\.atom\packages\linter-tslint\node_modules\tslint\lib\tslint.js:56:44)
    at file:///C:/Users/Artur/.atom/packages/linter-tslint/lib/init.coffee:103:31
```

**Comment by MaxMilton** [comment]
++

I got so many of these errors I had to disable linter-tslint.

**Comment by Maistho** [comment]
This is fixed in typescript master right now, but hasn't yet been released.

https://github.com/Microsoft/TypeScript/commit/4162897350209538a64941916aca6235e58daca4

Until then, you can open `node_modules/typescript/lib/typescript.js` and make the change yourself at around line 61632.

**Comment by ThomasHandwerker** [comment]
Worked for me! Thanks for the workaround.

**Comment by Xapphire13** [comment]
Looks like this has been fixed since TypeScript 2.1.
Closing since we are now using TypeScript 2.3.1

**Comment by Arcanemagus** [comment]
@Xapphire13 Not to mention, since the move to a worker process the workaround isn't required in the first place 😉.
